src/edu/hillel/lesson23/factory/FactoryI.java

```diff
@@ -0,0 +1,8 @@
+package edu.hillel.lesson23.factory;
+
+public interface FactoryI {
```

Comment:
Please, avoid literal I in the and of name. Better call it CommonFactory or AbstractFactory or just Factory